tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,516 @@
+    assert isinstance(response["id"], int)
+    assert isinstance(response["snapshotId"], int)
+    assert isinstance(response["name"], str)
+    assert isinstance(response["description"], str)
+    assert isinstance(response["createdOn"], str)
+    assert isinstance(response["lastModifiedOn"], str)
+    assert isinstance(response["latestSnapshot"], bool)
+
+    assert response["name"] == expected_contents["name"]
+    assert response["description"] == expected_contents["description"]
+
+    assert is_iso_format(response["createdOn"])
+    assert is_iso_format(response["lastModifiedOn"])
+
+    # Validate the UserRef structure
+    assert isinstance(response["user"]["id"], int)
+    assert isinstance(response["user"]["username"], str)
+    assert isinstance(response["user"]["url"], str)
+    assert response["user"]["id"] == expected_contents["user_id"]
+
+    # Validate the GroupRef structure
+    assert isinstance(response["group"]["id"], int)
+    assert isinstance(response["group"]["name"], str)
+    assert isinstance(response["group"]["url"], str)
+    assert response["group"]["id"] == expected_contents["group_id"]
+
+    # Validate the TagRef structure
+    for tag in response["tags"]:
+        assert isinstance(tag["id"], int)
+        assert isinstance(tag["name"], str)
+        assert isinstance(tag["url"], str)
+
+
+def assert_retrieving_queue_by_id_works(
+    client: FlaskClient,
+    queue_id: int,
+    expected: dict[str, Any],
+) -> None:
+    """Assert that retrieving a queue by id works.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}", follow_redirects=True
+    )
+    assert response.status_code == 200 and response.get_json() == expected
+
+
+# TODO handle Q,G,P
+def assert_retrieving_all_queues_works(
+    client: FlaskClient,
+    expected: list[dict[str, Any]],
+    Q: dict[str, Any] | None = None,
+    G: dict[str, Any] | None = None,
+    P: dict[str, Any] | None = None,
+) -> None:
+    """Assert that retrieving all queues works.
+
+    Args:
+        client: The Flask test client.
+        expected: The expected response from the API.
+        Q: The search query parameters.
+        G: The group query parameters.
+        P: The paging query parameters.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+
+    payload = {}
+
+    if Q:
+        payload.update(Q)
+    if G:
+        payload.update(G)
+    if P:
+        payload.update(P)
+
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}", payload=payload, follow_redirects=True
```

Comment:
I don't think there's a `payload` argument in the `.get` method, it's usually either `json=` (when passing a JSON serializable dict) or `data=` (which I think allows you to pass anything, not just dicts).

In the `requests` library, there is a `params` argument for the query parameters: https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls. I suspect the Flask Test Client has a similar field.

Reply:
looks like you can pass a dict as `query_string`: https://flask.palletsprojects.com/en/2.3.x/testing/